app/controllers/mixins/actions/vm_actions/ownership.rb

```diff
@@ -150,69 +124,97 @@ def valid_items_for(klass, param_ids)
 
         def ownership_update
           case params[:button]
-          when "cancel"
-            add_flash(_("Set Ownership was cancelled by the user"))
+          when "cancel" then ownership_handle_cancel_button
+          when "save" then ownership_handle_save_button
+          when "reset" then ownership_handle_reset_button
+          end
+        end
+
+        private
+
+        def load_user_group_items(ownership_ids, klass)
+          @ownershipitems ||= filter_ownership_items(klass, ownership_ids)
+          if ownership_ids.length > 1
+            @user = @group = 'dont-change'
+          else
+            record = @ownershipitems.first
+            @user = record.evm_owner.id.to_s if record.evm_owner
+            @group = record.miq_group.id.to_s if record.miq_group
+          end
+        end
+
+        def ownership_handle_cancel_button
+          add_flash(_("Set Ownership was cancelled by the user"))
+          if @sb[:explorer]
+            @edit = @sb[:action] = nil
+            replace_right_cell
+          else
+            session[:flash_msgs] = @flash_array
+            javascript_redirect previous_breadcrumb_url
+          end
+        end
+
+        def ownership_handle_save_button
+          opts = {}
+          opts[:owner] = unless params[:user] == 'dont-change'
```

Comment:
This breaks multi-target set ownership...

Before, `owner` or `group` only got assigned to if the user chose to clear the previously set owner/group.
Now, this is always nil, so it's impossible to change user without resetting groups, and vice versa.

Fixing in...